sound/soc/sof/intel/hda.c

```diff
@@ -185,6 +185,10 @@ static int hda_sdw_probe(struct snd_sof_dev *sdev)
 
 	memset(&res, 0, sizeof(res));
 
+	/* sdw trigger callback is only needed for IPC4 */
+	if (sdev->pdata->ipc_type == SOF_INTEL_IPC4)
```

Comment:
can we make it " static struct sdw_intel_ops sdw_callback[2] = {.....}

Reply:
> can we make it " static struct sdw_intel_ops sdw_callback[2] = {.....}

Do you mean sdw_callback[0] for IPC3 and sdw_callback[1] for IPC4?
What is the What is the benefit of it?

Reply:
Changed to `if` and added comment to explain why we set ALH blob here.

Reply:
> Follow ops array setting like topology or ipc. "if xxx == ipc4" likes a hack

I moved  `sdw_callback.trigger = ipc4_be_dai_common_trigger;`  to `hda_set_dai_drv_ops` in the SOF_INTEL_IPC4 case now.  BTW, we did the same thing in sof_xxx_ops_init to set topology ops :)